src/pages/rekey-reason/components/other-reason/other-reason.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, Input, Output, EventEmitter, OnChanges } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'other-reason',
+  templateUrl: 'other-reason.html',
+})
+export class OtherReasonComponent implements OnChanges {
+
+  static readonly checkBoxCtrl: string = 'otherSelected';
+  static readonly fieldName: string = 'reason';
+
+  @Input()
+  selected: boolean;
+
+  @Input()
+  reason: string;
+
+  @Input()
+  formGroup: FormGroup;
+
+  @Output()
+  selectedChange = new EventEmitter<boolean>();
+
+  @Output()
+  reasonChange = new EventEmitter<string>();
+
+  private checkBoxFormControl: FormControl;
+  private formControl: FormControl;
+
+  public reasonDescriptionCharsRemaining: number = null;
+
+  ngOnChanges(): void {
+    if (!this.checkBoxFormControl) {
+      this.checkBoxFormControl = new FormControl(null);
+      this.formGroup.addControl(OtherReasonComponent.checkBoxCtrl, this.checkBoxFormControl);
+    }
+
+    if (!this.formControl) {
+      this.formControl = new FormControl(null);
+      this.formGroup.addControl(OtherReasonComponent.fieldName, this.formControl);
+    }
+
+    if (this.selected) {
+      this.formGroup.get(OtherReasonComponent.fieldName).setValidators([
+        Validators.required, Validators.maxLength(200)]);
+    } else {
+      this.formGroup.get(OtherReasonComponent.fieldName).clearValidators();
+    }
+
+    this.checkBoxFormControl.patchValue(this.selected ? true : false);
+    this.formControl.patchValue(this.reason);
+  }
+
+  selectedValueChanged(selected: boolean): void {
+    if (!selected) {
+      this.formGroup.get(OtherReasonComponent.fieldName).reset();
+    }
+    this.selectedChange.emit(selected);
+  }
+
+  reasonTextChanged(reason: string): void {
+    this.reasonChange.emit(reason);
+  }
+
+  characterCountChanged(charactersRemaining: number) {
+    this.reasonDescriptionCharsRemaining = charactersRemaining;
+  }
+
+  getCharacterCountText() {
```

Comment:
May be me not understanding the character counter, but doesn't the directive do all of this stuff

Reply:
The directive emits the count, but currently the job of displaying the count and validation message is manual. There's work here for a future refactor.